src/libsyntax/parse/attr.rs

```diff
@@ -35,7 +35,10 @@ impl<'a> Parser<'a> {
                     self.span.hi
                 );
                     if attr.node.style != ast::AttrStyle::Outer {
-                        return Err(self.fatal("expected outer comment"));
+                        let mut err = self.fatal("expected outer comment");
+                        err.fileline_note(self.span, "inner comments like this must \
+                                                      be placed before any items");
```

Comment:
perhaps mention what an inner comment  is (or show examples, e.g. `//!..`)? That's pretty obscure terminology.